### PR TITLE
chore(core) Document: add test checking pageContents empty string behavior

### PR DESCRIPTION
**Pull request by skarard**
This PR is related to a discussion in #2807, the current behaviour of the Document constructor `field.pageContent` when given an empty string is to set `pageContent` as `undefined`. This broke `HNSWLib.addDocuments()` as it attempts a string.replace() on `undefined` rather than the typed `string` expected.

I suggested allowing `pageContents` to be set as an empty string. @jacoblee93 suggested that a Document with an empty `pageContents` should not be created at all.

This implementation requires `pageContents` to be a string that is not empty, otherwise it throws `Error("pageContents must not be empty")`.

### CODE REVIEW

**Comment by jacoblee93** [comment]
Ideally we shouldn't need this with TypeScript, but an empty string should be permissible, just not a hard `undefined`, `null`, or non-string value - can you change?

**Comment by skarard** [comment]
Sure I change this out to allow it to record an empty string.

**Comment by skarard** [comment]
@jacoblee93 I added the proposed change and threw in a couple of unit tests for good measure.

**Review by bracesproul — CHANGES_REQUESTED** [review]
Hey @skarard sorry for taking so long to review, these changes look good! We've moved `document.ts` into `langchain-core/src/documents/document.ts`, could you push up a change moving your changes into that updated file then re-request my review?

Thanks again!

File: langchain-core/src/documents/tests/document.test.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Document } from "../document.js";
```
Comment (benjamincburns) [inline langchain-core/src/documents/tests/document.test.ts:1]:
```suggestion
import { Document } from "../document.js";
import { describe, expect, test} from "@jest/globals";
```

File: langchain-core/src/documents/tests/document.test.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import { describe, expect, test} from "@jest/globals";
```
Comment (benjamincburns) [inline langchain-core/src/documents/tests/document.test.ts:2]:
```suggestion
import { describe, expect, test } from "@jest/globals";
```

**Comment by benjamincburns** [comment]
Thanks for the contribution!

**Review by benjamincburns — APPROVED** [review]
